src/Npgsql/NpgsqlConnector.cs

```diff
@@ -298,6 +298,9 @@ internal NpgsqlConnector(NpgsqlConnection connection)
             ConnectionString = connectionString;
             PostgresParameters = new Dictionary<string, string>();
             Transaction = new NpgsqlTransaction(this);
+            CommandCts = new TimeoutCancellationTokenSourceWrapper(settings.CancellationTimeout > 0
+                ? TimeSpan.FromSeconds(settings.CancellationTimeout)
+                : Timeout.InfiniteTimeSpan);
```

Comment:
Check this in the constructor.

Reply:
I don't think there's any real need to pass the timeout here. We set the timeout every time a command is executed (since it may have a different CommandTimeout), so this isn't really needed.